File: src/test/java/discretemath/structure/set/SetTest.java
package discretemath.structure.set;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class SetTest {

	@Test
	void union() {
		Set<Integer> set1 = SimpleSet.of(1, 2, 3);
		Set<Integer> set2 = SimpleSet.of(3, 4, 5);

		assertEquals(SimpleSet.of(1, 2, 3, 4, 5), set1.unionWith(set2));

	}

	@Test
	void intersect() {
		Set<Integer> set1 = SimpleSet.of(1, 2, 3, 4);
		Set<Integer> set2 = SimpleSet.of(3, 4, 5);

		assertEquals(SimpleSet.of(3, 4), set1.intersectWith(set2));
	}

	@Test
	void difference() {
		Set<Integer> set1 = SimpleSet.of(1, 2, 3, 4);
		Set<Integer> set2 = SimpleSet.of(3, 4, 5);

		assertEquals(SimpleSet.of(1, 2), set1.differenceWith(set2));
	}

	@Test
	void symmetricDifference() {
		Set<Integer> set1 = SimpleSet.of(1, 2, 3, 4);
		Set<Integer> set2 = SimpleSet.of(3, 4, 5);

		assertEquals(SimpleSet.of(1, 2, 5), set1.symmetricDifferenceWith(set2));
	}
}

File: src/test/scala/discretemath/test/structure/set/main/SetTest.scala
package discretemath.test.structure.set.main

import discretemath.structure.set.main.SimpleSet
import org.scalatest.funsuite.AnyFunSuite;


class SetTest extends AnyFunSuite {

  test("union") {
    val set1 = SimpleSet(1, 2, 3);
    val set2 = SimpleSet(3, 4, 5);

    assert(set1.unionWith(set2) == SimpleSet(1, 2, 3, 4, 5));
  }

  test("intersect") {
    val set1 = SimpleSet(1, 2, 3, 4);
    val set2 = SimpleSet(3, 4, 5);

    assert(set1.intersectWith(set2) == SimpleSet(3, 4));
  }

  test("difference") {
    val set1 = SimpleSet(1, 2, 3, 4);
    val set2 = SimpleSet(3, 4, 5);

    assert(set1.differenceWith(set2) == SimpleSet(1, 2));
  }

  test("symmetricDifference") {
    val set1 = SimpleSet(1, 2, 3, 4);
    val set2 = SimpleSet(3, 4, 5);

    assert(set1.symmetricDifferenceWith(set2) == SimpleSet(1, 2, 5));
  }
}